core/stdlib/std.ncl

```diff
@@ -792,6 +792,56 @@
   },
 
   contract = {
+    Assert
+      | doc m%"
+        A contract that checks if its argument is the boolean true.
+
+        We could get by just using arrays of boolean tests and `std.array.all`.
+
+        However, this contract provides fine-grained error reporting,
+        pinpointoing the exact expression that failed in an array of tests, as
+        opposed to the pure boolean solution.
+
+        # Examples
+
+        ```nickel
+        1 == 2 | std.contract.Assert
+         => error: contract broken by a value
+
+        1 == 1 | std.contract.Assert
+         => true
+        ```
+        "%
+        = fun l x => if x then true else %blame% l,
+
+    # We can't use a static type because x | Assert is not of type bool. We
+    # could use additional contracts to make it work but it's not worth the
+    # hassle.
+    assert_all
+      | doc m%"
+
+        Assert an array of boolean statements are true. Apply the
+        `std.contract.Assert` contract on each element of the given array.
+
+        `std.contract.assert_all tests` is semantically equivalent to
+
+        ```nickel
+        (std.array.all function.id tests) | std.contract.Assert
```

Comment:
Initially I just wanted  to add  the missing `std` in front of `function.id`, but `std.function.id` is just longer than `fun x => x` :man_shrugging: 

```suggestion
        (std.array.all (fun x => x) tests) | std.contract.Assert
```